File: src/main/kotlin/com/example/kotlin/config/DataSourceConfiguration.java
package com.example.kotlin.config;

import ch.vorburger.exec.ManagedProcessException;
import ch.vorburger.mariadb4j.DBConfigurationBuilder;
import ch.vorburger.mariadb4j.springframework.MariaDB4jSpringService;
import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
import org.springframework.boot.jdbc.DataSourceBuilder;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.DependsOn;

import javax.sql.DataSource;

@Configuration
public class DataSourceConfiguration {

    @Bean
    MariaDB4jSpringService mariaDB4jSpringService() {
        return new MariaDB4jSpringService();
    }

    @Bean
    @DependsOn("mariaDB4jSpringService")
    DataSource dataSource(MariaDB4jSpringService mariaDB4jSpringService, DataSourceProperties dataSourceProperties) throws ManagedProcessException {

        DBConfigurationBuilder config = mariaDB4jSpringService.getConfiguration();

        return DataSourceBuilder.create()
                .username(dataSourceProperties.getUsername())
                .password(dataSourceProperties.getPassword())
                .url(dataSourceProperties.getUrl())
                .driverClassName(dataSourceProperties.getDriverClassName())
                .build();
    }

}

File: src/main/kotlin/com/example/kotlin/config/DataSourceConfiguration.kt
package com.example.kotlin.config

import ch.vorburger.mariadb4j.DBConfigurationBuilder
import ch.vorburger.mariadb4j.springframework.MariaDB4jSpringService
import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties
import org.springframework.boot.jdbc.DataSourceBuilder
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.DependsOn
import javax.sql.DataSource

@Configuration
class DataSourceConfiguration {

    @Bean("mariaDB4jSpringService")
    fun mariaDb4jSpringService(): MariaDB4jSpringService = MariaDB4jSpringService()

    @Bean
    @DependsOn("mariaDB4jSpringService")
    fun dataSource(mariaDB4jSpringService: MariaDB4jSpringService, dataSourceProperties: DataSourceProperties): DataSource {

        val config: DBConfigurationBuilder = mariaDB4jSpringService.configuration

        return DataSourceBuilder.create()
            .username(dataSourceProperties.username)
            .password(dataSourceProperties.password)
            .url(dataSourceProperties.url)
            .driverClassName(dataSourceProperties.driverClassName)
            .build()

    }

}